substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/elf/dwarf/DwarfSections.java

```diff
@@ -0,0 +1,2344 @@
+    }
+
+    public FileEntry ensureFileEntry(Range range) {
+        String fileName = range.getFileName();
+        // ensure we have an entry
+        FileEntry fileEntry = filesIndex.get(fileName);
+        if (fileEntry == null) {
+            DirEntry dirEntry = ensureDirEntry(fileName);
+            String baseName = (dirEntry == null ? fileName : fileName.substring(dirEntry.getPath().length() + 1));
+            fileEntry = new FileEntry(stringTable.uniqueDebugString(fileName),
+                            stringTable.uniqueString(baseName),
+                            dirEntry);
+            files.add(fileEntry);
+            filesIndex.put(fileName, fileEntry);
+            // if this is a primary entry then add it to the primary list
+            if (range.isPrimary()) {
+                primaryFiles.add(fileEntry);
+            } else {
+                Range primaryRange = range.getPrimary();
+                FileEntry primaryEntry = filesIndex.get(primaryRange.getFileName());
+                assert primaryEntry != null;
+            }
+        }
+        return fileEntry;
+    }
+
+    public void addRange(Range primaryRange, List<DebugFrameSizeChange> frameSizeInfos, int frameSize) {
+        assert primaryRange.isPrimary();
+        ClassEntry classEntry = ensureClassEntry(primaryRange);
+        PrimaryEntry entry = classEntry.addPrimary(primaryRange, frameSizeInfos, frameSize);
+        if (entry != null) {
+            // track the entry for this range in address order
+            primaryEntries.add(entry);
+        }
+    }
+
+    public void addSubRange(Range primaryRange, Range subrange) {
+        assert primaryRange.isPrimary();
+        assert !subrange.isPrimary();
+        String className = primaryRange.getClassName();
+        ClassEntry classEntry = primaryClassesIndex.get(className);
+        FileEntry subrangeEntry = ensureFileEntry(subrange);
+        // the primary range should already have been seen
+        // and associated with a primary class entry
+        assert classEntry.primaryIndexFor(primaryRange) != null;
+        classEntry.addSubRange(subrange, subrangeEntry);
+    }
+
+    private DirEntry ensureDirEntry(String file) {
+        int pathLength = file.lastIndexOf('/');
+        if (pathLength < 0) {
+            // no path/package means use dir entry 0
+            return null;
+        }
+        String filePath = file.substring(0, pathLength);
+        DirEntry dirEntry = dirsIndex.get(filePath);
+        if (dirEntry == null) {
+            dirEntry = new DirEntry(stringTable.uniqueString(filePath));
+            dirsIndex.put(filePath, dirEntry);
+            dirs.add(dirEntry);
+        }
+        return dirEntry;
+    }
+
+    /**
+     * class from which all DWARF debug section
+     * inherit providing common behaviours.
+     */
+    // shared implementation methods to manage content creation
+    public abstract class DwarfSectionImpl extends BasicProgbitsSectionImpl {
```

Comment:
Should be a static class. Others likely as well. Event better split out into several source files

Reply:
Fixed as part of the move to separate files.